osquery/tables/system/posix/yum_sources.cpp

```diff
@@ -0,0 +1,93 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <boost/property_tree/ini_parser.hpp>
+#include <iostream>
+#include <fstream>
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string kYumConf { "/etc/yum.conf" };
+const std::string kYumReposDir { "/etc/yum.repos.d" };
+
+void parseYumConf(std::istream& source, QueryData& results, std::string& repos_dir) {
+  boost::property_tree::ptree tree;
+  boost::property_tree::ini_parser::read_ini(source, tree);
+  repos_dir = tree.get("main.reposdir", kYumReposDir);
+
+  for (auto it1: tree) {
+    // Section
+    if (it1.first == "main") {
+      continue;
+    }
+
+    Row r;
+    for (auto it2: it1.second) {
+      // Option
+      if ("baseurl" == it2.first || "enabled" == it2.first
+          || "gpgcheck" == it2.first || "name" == it2.first
+          || "gpgkey" == it2.first) {
+        r[it2.first] = it2.second.data();
+      }
+    }
+    results.push_back(r);
+  }
+}
+
+void parseYumConf(const std::string& source, QueryData& results, std::string& repos_dir) {
+  std::ifstream stream(source.c_str());
+  if (!stream) {
+    VLOG(1) << "File " << source
+      << " either cannot be read or cannot be parsed as ini";
```

Comment:
Please be accurate with the error here and later.

Reply:
I'll correct it here but at the other spot that's about as accurate as you get. ini_parser mangles IO error and parse error under ini_parser_error so it's impossible to tell which one is which